src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.Keyboard.cs

```diff
@@ -0,0 +1,312 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Gdk;
+using Gtk;
+using Microsoft.Extensions.Logging;
+using Uno.Extensions;
+using Uno.Foundation.Extensibility;
+using Uno.Logging;
+using Uno.UI.Runtime.Skia;
+using Uno.UI.Runtime.Skia.GTK.Extensions;
+using Windows.ApplicationModel;
+using Windows.Devices.Input;
+using Windows.Foundation;
+using Windows.System;
+using Windows.UI.Composition;
+using Windows.UI.Core;
+using Windows.UI.Input;
+
+namespace Uno.UI.Runtime.Skia
+{
+	public partial class GtkCoreWindowExtension : ICoreWindowExtension
+	{
+		partial void InitializeKeyboard()
+		{
+			Gtk.Key.SnooperInstall(OnKeySnoop);
+		}
+
+		private int OnKeySnoop(Widget grab_widget, EventKey e)
+		{
+			if(e.Type == EventType.KeyPress)
+			{
+				OnKeyPressEvent(e);
+			}
+			else if(e.Type == EventType.KeyRelease)
+			{
+				OnKeyReleaseEvent(e);
+			}
+
+			return 1;
+		}
+
+		private void OnKeyPressEvent(EventKey evt)
+		{
+			try
+			{
+				var virtualKey = ConvertKey(evt.Key);
+
+				if (this.Log().IsEnabled(LogLevel.Trace))
+				{
+					this.Log().Trace($"OnKeyPressEvent: {evt.Key} -> {virtualKey}");
+				}
+
+				_ownerEvents.RaiseKeyDown(
+					new KeyEventArgs(
+						"keyboard",
+						virtualKey,
+						new CorePhysicalKeyStatus
+						{
+							ScanCode = evt.HardwareKeycode,
+							RepeatCount = 1,
+						}));
+			}
+			catch (Exception e)
+			{
+				if (this.Log().IsEnabled(LogLevel.Error))
+				{
+					this.Log().Error("Failed to raise KeyDown", e);
+				}
+			}
+		}
+
+		private void OnKeyReleaseEvent(EventKey evt)
+		{
+			try
+			{
+				var virtualKey = ConvertKey(evt.Key);
+
+				if (this.Log().IsEnabled(LogLevel.Trace))
+				{
+					this.Log().Trace($"OnKeyReleaseEvent: {evt.Key} -> {virtualKey}");
+				}
+
+				_ownerEvents.RaiseKeyUp(
+					new KeyEventArgs(
+						"keyboard",
+						virtualKey,
+						new CorePhysicalKeyStatus
+						{
+							ScanCode = evt.HardwareKeycode,
+							RepeatCount = 1,
+						}));
+			}
+			catch(Exception e)
+			{
+				this.Log().Error("Failed to raise KeyUp", e);
+			}
+		}
+
+		private VirtualKey ConvertKey(Gdk.Key key)
+		{
+			return key switch
+			{
+				Gdk.Key.VoidSymbol => VirtualKey.None,
+
+				// Gdk.Key.Left => VirtualKey.LeftButton,
```

Comment:
should the commented entries be committed? If so maybe add a comment indicating why